node.flat.js

```diff
@@ -0,0 +1,3 @@
+const createNodeConfig = require('./node.factory');
+
+module.exports = createNodeConfig(true);
```

Comment:
instead of a boolean flag in an argument that controls whether this is a flat config or not, can we have two specialized functions instead of one? `createFlatNodeConfig` and `createLegacyNodeConfig` or something that makes sense semantically?

Reply:
Sure, these names make sense - refactored in 10889e920cc69a00da7ce21b71e06570e4cf1744